src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1311,7 +1311,7 @@ private BoundExpression GetDefaultParameterValue(SyntaxNode syntax, ParameterSym
                 BoundExpression memberNameLiteral = MakeLiteral(syntax, ConstantValue.Create(path), compilation.GetSpecialType(SpecialType.System_String), localRewriter);
                 defaultValue = MakeConversionNode(memberNameLiteral, parameterType, @checked: false);
             }
-            else if (parameter.IsCallerMemberName && ((callerSourceLocation = GetCallerLocation(syntax, enableCallerInfo)) != null))
+            else if (parameter.IsCallerMemberName && ((_ = GetCallerLocation(syntax, enableCallerInfo)) != null))
```

Comment:
surprising!

Reply:
confirmed it is actually a true violation as `callerSourceLocation` is not used later on this code path.

Reply:
If the assignment is not in a statement, would it make sense to always just leave off the discard and just use the initializer expression?